src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp

```diff
@@ -344,8 +345,13 @@ bool FlightTaskAuto::_evaluateTriplets()
 
 	_type = (WaypointType)_sub_triplet_setpoint.get().current.type;
 
-	// Always update cruise speed since that can change without waypoint changes.
-	_mc_cruise_speed = _sub_triplet_setpoint.get().current.cruising_speed;
+	// Override a commanded cruise speed if the cruise speed from the triplet is valid and more recent
+	const float cruise_speed_from_triplet = _sub_triplet_setpoint.get().current.cruising_speed;
+
+	if (PX4_ISFINITE(cruise_speed_from_triplet) && cruise_speed_from_triplet >= 0
```

Comment:
```suggestion
	if (PX4_ISFINITE(cruise_speed_from_triplet) && cruise_speed_from_triplet > 0.f
```
Does a cruise speed == 0.f make sense and is it supported?

Reply:
I think we should allow a very small speed, but probably not quite 0, so this change makes sense to me.